app/src/main/java/de/triology/cas/services/dogu/CesDoguServiceFactory.java

```diff
@@ -12,25 +12,47 @@
     public RegexRegisteredService createCASService(long id, String fqdn){
         RegexRegisteredService casService = new RegexRegisteredService();
         casService.setId(id);
-        casService.setServiceId("https://" + fqdn + "/cas/.*");
+        casService.setServiceId("https://" + generateServiceIdFqdnRegex(fqdn) + "/cas/.*");
         casService.setName(CesDoguServiceFactory.class.getSimpleName() + " " + SERVICE_CAS_IDENTIFIER);
         return casService;
     }
 
     @Override
     public RegexRegisteredService createNewService(long id, String fqdn, URI casLogoutURI, CesServiceData serviceData) throws CesServiceCreationException {
+        String fqdnRegex = generateServiceIdFqdnRegex(fqdn);
+
         RegexRegisteredService service = new RegexRegisteredService();
         service.setId(id);
 
-        String serviceId = "https://" + fqdn + "(:443)?/" + serviceData.getName() + "(/.*)?";
+        String serviceId = "https://" + fqdnRegex + "(:443)?/" + serviceData.getName() + "(/.*)?";
         service.setServiceId(serviceId);
         service.setName(serviceData.getIdentifier());
 
         if (casLogoutURI != null) {
-            String logoutUri = "https://" + fqdn + "/" + serviceData.getName() + casLogoutURI;
+            String logoutUri = "https://" + fqdnRegex + "/" + serviceData.getName() + casLogoutURI;
```

Comment:
I think this should not be a regex. Furthermore unit/ integration tests seem to be missing here, if this is not a problem.